libraries/base/java-utils/Utils.java

```diff
@@ -123,6 +123,23 @@ public static String byteBufferToStr(ByteBuffer buffer)
         return new String(bytes, "UTF-8");
     }
 
+    public static ByteBuffer c_memcpy(ByteBuffer dest, ByteBuffer src, int size) {
+        src = src.duplicate();
+        ByteBuffer dest2 = dest.duplicate();
+        src.limit(src.position() + size);
+        dest2.put(src);
+        return dest;
```

Comment:
Return `dest2` instead of `dest`. Always return new objects to keep all the operations pure. That's why we do a lot of `duplicate()`'s when dealing with `ByteBuffer`'s. This may not be a problem, but it's good to stay safe.